stan/math/prim/fun/norm1.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef STAN_MATH_PRIM_FUN_NORM1_HPP
+#define STAN_MATH_PRIM_FUN_NORM1_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns L1 norm of a vector. For vectors that equals the
+ * sum of magnitudes of its individual elements.
+ *
+ * @tparam T type of the vector (must be derived from \c Eigen::MatrixBase)
+ * @param v Vector.
+ * @return L1 norm of v.
+ */
+template <typename Container, require_container_t<Container>* = nullptr,
+          require_not_st_var<Container>* = nullptr,
```

Comment:
These templates can be a little stricter by specifying that they're only for arithmetic types, see the [`log_softmax` header for an example](https://github.com/stan-dev/math/blob/83dbd82c5b431d22c6e5899b2dfc5830a4cf95ed/stan/math/prim/fun/log_softmax.hpp#L42)

Reply:
Cool, I changed the templates to include `require_st_arithmetic<>` to match the template of `log_softmax`. 

Does `require_st_arithmetic<>` do the job of both `require_not_st_var<>` and `require_not_st_fvar<>`? These are what I used to resolve ambiguity between `prim` vs `rev` and `prim` vs `fwd`, respectively.

Reply:
Yep, it will restrict to only matching inputs that satisfy `std::is_arithmetic`